### PR TITLE
Merge `vssr_pourbaix`

### PR DESCRIPTION
**Pull request by xiaochendu**
Additional and improved methods for fine-tuning and evaluating/plotting NFF

### CODE REVIEW

**Review by pafervi — DISMISSED** [review]

**Review by ajhoffman1229 — COMMENTED** [review]
I think most of the code looks good, there are a few places where it looks like there are some duplicate lines though.

File: nff/data/stats.py [hunk 1]
```diff
@@ -48,13 +48,21 @@ def remove_outliers(
...
-    mean = reference_mean if reference_mean else np.mean(stats_array)
-    std = reference_std if reference_std else np.std(stats_array)
+    if reference_mean is None:
+        mean = np.mean(stats_array)
+    else:
+        mean = reference_mean
+    if reference_std is None:
+        std = np.std(stats_array)
+    else:
+        std = reference_std
```
[the remark this answers is not in the document]
Comment (xiaochendu) [reply nff/data/stats.py:65]: I must have made an error when merging. I wanted to take the incoming (master) rather than current (vssr_pourbaix).

File: nff/io/ase_calcs.py [hunk 2]
```diff
@@ -132,6 +132,8 @@ def calculate(
             self.properties = properties
 
         Calculator.calculate(self, atoms, self.properties, system_changes)
+        # TODO: update atoms only when necessary
+        atoms.update_nbr_list(update_atoms=True)
```
[the remark this answers is not in the document]
Comment (xiaochendu) [reply nff/io/ase_calcs.py:136]: This is mostly done to update the number of atoms in between MCMC steps. It might not be that necessary for the general user running MD. Let me remove it for the main branch.

File: nff/utils/cuda.py [hunk 3]
```diff
@@ -114,5 +114,9 @@ def get_final_device(device: str) -> str:
         str: final device to use
     """
     if "cuda" in device and torch.cuda.is_available():
+        try:
+            return f"cuda:{cuda_devices_sorted_by_free_mem()[-1]}"
+        except nvidia_smi.NVMLError:
+            return "cuda:0"
         return f"cuda:{cuda_devices_sorted_by_free_mem()[-1]}"
```
[the remark this answers is not in the document]
Comment (xiaochendu) [reply nff/utils/cuda.py]: you're right!

[the remark this answers is not in the document]
Comment (xiaochendu) [reply nff/utils/cuda.py]: Good catch

**Review by ajhoffman1229 — DISMISSED** [review]
Looks great!

**Review by HojeChun — APPROVED** [review]

File: nff/io/chgnet.py [hunk 4]
```diff
@@ -59,39 +51,135 @@ def convert_nff_to_chgnet_structure_data(
...
+        "formula": formula,
+        "units": units,
+    }
+    return Dataset(concated_batch, units=units[0])
+
+
 def convert_data_batch(
     data_batch: Dict,
     cutoff: float = 5.0,
     shuffle: bool = True,
```
Comment (HojeChun) [inline nff/io/chgnet.py:163]: Can we set shuffle to be False. I assume shuffle has been done when you make dataloader, and this is a function for wrapper.

**Review by HojeChun — COMMENTED** [review]
Looks great to me just had one suggestion.

**Review by HojeChun — APPROVED** [review]
